text/0814-cdk-mixins.md

```diff
@@ -0,0 +1,758 @@
+import { CfnLogGroup } from "aws-cdk-lib/aws-logs";
+
+// Apply encryption to all log groups in the scope
+Mixins.of(
+  scope,
+  ConstructSelector.resourcesOfType(cloudwatch.CfnLogGroup)
+).apply(new EncryptionAtRest());
+```
+
+#### Types of Mixins
+
+**Resource-Specific Mixins**: Tailored to individual AWS services
+
+```typescript
+// Hand-written mixins for common patterns
+const bucket = new s3.CfnBucket(scope, "Bucket")
+  .with(new BucketEvents())
+  .with(new BucketGrants());
+
+// Auto-generated mixins from AWS service specifications
+const bucketWithNewFeature = new s3.CfnBucket(scope, "NewBucket")
+  .with(new CfnBucketPropsMixin({ 
+    someNewFeatureConfig: { settingOne: "value1" } 
+  }));
+```
+
+**Cross-Service Mixins**: Common patterns across AWS services
+
+```typescript
+// Same mixin works across different resource types
+const bucket = new s3.CfnBucket(scope, "Bucket")
+  .with(new EncryptionAtRest());
+
+const logGroup = new logs.CfnLogGroup(scope, "LogGroup")
+  .with(new EncryptionAtRest());
+
+const table = new dynamodb.CfnTable(scope, "Table")
+  .with(new EncryptionAtRest());
+```
+
+#### Using Mixins with L1 Constructs
+
+Mixins unlock the full potential of L1 constructs by adding sophisticated abstractions:
+
+```typescript
+// L1 with enterprise-grade features
+const bucket = new s3.CfnBucket(scope, "EnterpriseBucket")
+  .with(new EncryptionAtRest())
+  .with(new AutoDeleteObjects())
+  .with(new ComplianceLogging())
+  .with(new CostOptimization());
+
+// Access day-one AWS features with abstractions
+const bucketWithLatestFeature = new s3.CfnBucket(scope, "LatestBucket")
+  .with(new CfnBucketPropsMixin({ 
+    // New CloudFormation property available immediately
+    newAwsFeature: { enabled: true }
+  }))
+  .with(new EncryptionAtRest());
+
+// Helper classes provide L2-like convenience
+new BucketGrants(bucket).grantRead(role);
+const eventPattern = new BucketEvents(bucket).onObjectCreated();
+```
+
+#### Using Mixins with L2 Constructs
+
+Mixins extend L2 constructs with additional capabilities while preserving existing functionality:
+
+```typescript
+// Existing L2 usage continues to work unchanged
+const standardBucket = new s3.Bucket(scope, "StandardBucket", {
+  autoDeleteObjects: true,
+  encryption: s3.BucketEncryption.S3_MANAGED
+});
+
+// Add new capabilities with mixins
+const enhancedBucket = new s3.Bucket(scope, "EnhancedBucket", {
+  autoDeleteObjects: true
+}).with(new ComplianceAuditing())
+  .with(new CostOptimization())
+  .with(new CfnBucketPropsMixin({ 
+    // Access new CloudFormation features not yet in L2
+    newFeature: { enabled: true }
+  }));
+
+// Mix L2 properties with mixin capabilities
+const hybridBucket = new s3.Bucket(scope, "HybridBucket", {
+  versioned: true,
+  lifecycleRules: [/* ... */]
+}).with(new AdvancedMonitoring())
+  .with(new SecurityCompliance());
+```
+
+#### Mixins and Aspects
```

Comment:
should we also talk abt its comparision to property injections? i know they are applied in different ways and serving mostly different purpose but isnt there overlap use cases?